src/bin/autograder_cli.rs

```diff
@@ -105,8 +105,29 @@ fn moodle_build(
             &task.as_ref().unwrap()
         );
         let result = read_check_toml(path.into_os_string().as_os_str())?;
-        let uuid = Uuid::now_v7();
-        build_task(&result, task.unwrap(), uuid)
+
+        if number > 1 {
+            // Multithread the task many times
+            let mut handles = vec![];
+
+            let config = Arc::new(&result);
+
+            for _i in 0..number {
+                let handle = thread::spawn(move || {});
+                let courseconf = Arc::clone(&config);
+                let uuid = Uuid::now_v7();
+                build_task(&courseconf, task.clone().unwrap(), uuid);
```

Comment:
I think this should be moved into `thread::spawn` before any multi threading occurs, like
```rust
let handle = thread::spawn(move || {
                let uuid = Uuid::now_v7();
                build_task(&config, task, uuid);
            });
```

Reply:
Changed it.